XenAdmin/Diagnostics/Problems/VMProblem/AutoStartEnabled.cs

```diff
@@ -95,7 +96,12 @@ private Action<Session> ActionDelegate(bool autostartValue)
                            {
                                vm.Locked = false;
                            }
-
+                           int wait = 1000; // wait up to 1 second for the cache to be updated

```

Comment:
I think this looks good, but I think a longer timeout like 5 seconds would be better to be more prepared for slow connections. Note that normally this code would be finished very quickly regardless of the (max)wait value.

Reply:
Agree. If the connection is really so slow, a waiting up to 5 seconds doesn't make things worse.